services/ingestion/event_subscriber.go

```diff
@@ -59,58 +72,59 @@ func NewRPCSubscriber(
 // to listen all new events in the current spork.
 //
 // If error is encountered during backfill the subscription will end and the response chanel will be closed.
-func (r *RPCSubscriber) Subscribe(ctx context.Context, height uint64) <-chan models.BlockEvents {
-	events := make(chan models.BlockEvents)
+func (r *RPCEventSubscriber) Subscribe(ctx context.Context) <-chan models.BlockEvents {
+	eventsChan := make(chan models.BlockEvents)
 
 	go func() {
 		defer func() {
-			close(events)
+			close(eventsChan)
 		}()
 
-		// if the height is from the previous spork, backfill all the events from previous sporks first
-		if r.client.IsPastSpork(height) {
+		// if the height is from the previous spork, backfill all the eventsChan from previous sporks first
+		if r.client.IsPastSpork(r.height) {
 			r.logger.Info().
-				Uint64("height", height).
+				Uint64("height", r.height).
 				Msg("height found in previous spork, starting to backfill")
 
-			// backfill all the missed events, handling of context cancellation is done by the producer
-			for ev := range r.backfill(ctx, height) {
-				events <- ev
+			// backfill all the missed eventsChan, handling of context cancellation is done by the producer
+			for ev := range r.backfill(ctx, r.height) {
+				eventsChan <- ev
 
 				if ev.Err != nil {
 					return
 				}
 
 				// keep updating height, so after we are done back-filling
 				// it will be at the first height in the current spork
-				height = ev.Events.CadenceHeight()
+				r.height = ev.Events.CadenceHeight()
 			}
 
 			// after back-filling is done, increment height by one,
 			// so we start with the height in the current spork
-			height = height + 1
+			r.height = r.height + 1
 		}
 
 		r.logger.Info().
-			Uint64("next-height", height).
+			Uint64("next-height", r.height).
 			Msg("backfilling done, subscribe for live data")
 
 		// subscribe in the current spork, handling of context cancellation is done by the producer
-		for ev := range r.subscribe(ctx, height, access.WithHeartbeatInterval(r.heartbeatInterval)) {
-			events <- ev
+		// TODO(JanezP): I think the heartbeat interval should always be 1 here
```

Comment:
You are right, this should actually always be set to `1`. But since the EVM block production happens on the system chunk transaction, we get a response from the Event Streaming API, for every block.
Right now, there's a config value for it, but I am not sure at what value it is set, for the instances on the various networks.

Reply:
removed the setting altogether. Please check again